src/Collections/Sites.php

```diff
@@ -140,6 +140,19 @@ public function filterByTag($tag)
         });
     }
 
+    /**
+     * Filters sites list by upstream
+     *
+     * @param string $upstream An upstream to filter by
+     * @return Sites
+     */
+    public function filterByUpstream($upstream_uuid)
+    {
+        return $this->filter(function ($model) use ($upstream_uuid) {
+            return ($model->getUpstream()->get('product_id') == $upstream_uuid);
```

Comment:
I think this would be more flexible if you use
```
in_array($upstream_id, $model->getUpstream()->getReferences())
```
And maybe also add the `product_id` attribute to its returned data here:
https://github.com/pantheon-systems/terminus/blob/master/src/Models/Upstream.php#L35
That way if the identifiers ever change or expand there'll be less code to change.

Reply:
This looks great and is probably the better solution, I was just never able to have getReferences() return anything other than an array of length 1 with the uuid.

Reply:
That's because you're accessing the upstream via its site. This model is also used for listing upstreams which involves accessing one or two different API. I linked the two in order to be able to use the same upstream-based operations from either approach.

Reply:
I considered making an abstract upstream and then making UserUpstream/OrgUpstream/SiteUpstream but really they're all the same thing.

Reply:
Ok I've added the proposed changes